cmd/node/node.go

```diff
@@ -600,28 +597,25 @@ func (app *App) initServices(ctx context.Context,
 
 	rabbit := app.HareFactory(ctx, mdb, sgn, nodeID, patrol, newSyncer, msh, tBeacon, hOracle, idStore, clock, lg)
 
-	stateAndMeshProjector := pendingtxs.NewStateAndMeshProjector(processor, msh)
-	minerCfg := miner.Config{
-		DBPath:         filepath.Join(dbStorepath, "builder"),
-		MinerID:        nodeID,
-		AtxsPerBlock:   app.Config.AtxsPerBlock,
-		LayersPerEpoch: layersPerEpoch,
-		TxsPerBlock:    app.Config.TxsPerBlock,
-	}
-
-	blockProducer := miner.NewBlockBuilder(
-		minerCfg,
+	proposalBuilder := miner.NewProposalBuilder(
+		ctx,
+		clock.Subscribe(),
 		sgn,
+		vrfSigner,
+		atxDB,
 		app.host,
-		clock.Subscribe(),
 		msh,
 		trtl,
-		blockOracle,
 		tBeacon,
 		newSyncer,
-		stateAndMeshProjector,
+		pendingtxs.NewStateAndMeshProjector(processor, msh).GetProjection,
```

Comment:
using interface has a benefit that you can find all implementations of the interface with a single command in IDE's

Reply:
changed